src/MUIDataTable.js

```diff
@@ -347,15 +360,18 @@ class MUIDataTable extends React.Component {
    *  Build the table data used to display to the user (ie: after filter/search applied)
    */
   computeDisplayRow(columns, row, rowIndex, filterList, searchText) {
+    let rowDataSource = this.props.options.rowDataSource;
     let isFiltered = false;
     let isSearchFound = false;
     let displayRow = [];
 
-    for (let index = 0; index < row.length; index++) {
-      let columnDisplay = row[index];
-      let columnValue = row[index];
+    displayRow = rowDataSource === "object" ? {} : [];
 
-      if (columns[index].customBodyRender) {
+    Object.keys(row).forEach((key, index) => {
```

Comment:
That will probably not work reliably as the order of properties on an object is not defined.
https://stackoverflow.com/questions/30076219/does-es6-introduce-a-well-defined-order-of-enumeration-for-object-properties

That means, that the index might not match to the provided column options object.

Reply:
Note: The order is same within one implementation of a javascript engine, eg. in Chrome it might always work.
But in FireFox, Safari, Edge or IE it will be different for each.

Thats why this approach seems to be working during development, but is not cross browser compatible

Reply:
You can use this code to create an array with an defined order from your object

´´´javascript
const dataObjects = [{prop1:1,prop2:2,prop3:3},{prop1:1,prop2:2,prop3:3}]
const dataArray = dataObjects.reduce<any[]>((all, data) => {
      all.push([
        data.prop1,
        data.prop2,
        data.prop3,
      ]);
      return all;
    }, []);
´´´

This then simply work with the current API

Reply:
Yes, you are right we can covert the array of object as you suggested. But I have also fix my implementation without using Object.keys. 
Please have a look